### PR TITLE
refactor: Rearrange impl.cpp as the same order as impl.h

### LINKED ISSUE #267: Generate intrinsics coverage report
As mentioned in https://github.com/DLTcollab/sse2neon/pull/263#issuecomment-755365873, we are going to generate intrinsics coverage reports with the result of CI.

The report can be appended [here](https://github.com/DLTcollab/sse2neon/blob/master/tests/main.cpp#L28)

### PR DESCRIPTION
**Pull request by HowJMay**
Rearrange impl.cpp as the same order as impl.h.

close #267

### CODE REVIEW

File: tests/main.cpp [hunk 1]
```diff
@@ -25,8 +25,12 @@ int main(int /*argc*/, const char ** /*argv*/)
         }
     }
     test->release();
-    printf("SSE2NEONTest Complete: Passed %d tests : Failed %d : Ignored %d\n",
-           passCount, failedCount, ignoreCount);
+    printf(
+        "SSE2NEONTest Complete: Passed %d tests : Failed %d : Ignored %d. "
+        "Coverage rate: %.2f%\n",
+        passCount, failedCount, ignoreCount,
+        (float) (passCount + failedCount) /
```
Comment (marktwtn) [inline tests/main.cpp:32]: I think we should not include `failedCount` here.

**Review by marktwtn — CHANGES_REQUESTED** [review]
The subject of the git commit message is weird.
Please rewrite the sentence `Arrange impl.cpp in the as impl.h`.

**Review by jserv — CHANGES_REQUESTED** [review]
Rewrite the subject. You should address the motivation and purpose of this change.

**Comment by HowJMay** [comment]
> Rewrite the subject. You should address the motivation and purpose of this change.

I have separated the coverage report to another PR

**Comment by marktwtn** [comment]
Rewrite the subject `Arrange impl.cpp in the as impl.h` before you requested my review.

**Comment by HowJMay** [comment]
> Rewrite the subject `Arrange impl.cpp in the as impl.h` before you requested my review.
> The `in the as` makes no sense.

done

**Comment by marktwtn** [comment]
> > Rewrite the subject `Arrange impl.cpp in the as impl.h` before you requested my review.
> > The `in the as` makes no sense.
> 
> done

The commit itself is not handled.

**Review by marktwtn — APPROVED** [review]

**Pull request merged**